### PR TITLE
9ci/domain9#2012 - error handler for oauth token controller

### PR DESCRIPTION
**Pull request by snimavat**
9ci/domain9#2012

### CODE REVIEW

File: security/boot-security/src/main/groovy/yakworks/security/spring/token/TokenController.groovy [hunk 1]
```diff
@@ -127,4 +132,11 @@ class TokenController {
             .body(body)
     }
 
+    @ExceptionHandler(Exception.class)
+    def handleException(HttpServletRequest req, HttpServletResponse resp, Exception e) {
+        log.error(e.message)
+        return [status: HttpStatus.INTERNAL_SERVER_ERROR.code, error:e.message]
```
Comment (basejump) [inline security/boot-security/src/main/groovy/yakworks/security/spring/token/TokenController.groovy:138]: dont break from our standards. should follow the problem api always.

**Review by basejump — CHANGES_REQUESTED** [review]
see

File: security/boot-security/src/main/groovy/yakworks/security/spring/token/TokenController.groovy [hunk 2]
```diff
@@ -127,4 +133,11 @@ class TokenController {
             .body(body)
     }
 
+    @ExceptionHandler(Exception.class)
+    def handleException(HttpServletRequest req, HttpServletResponse resp, Exception e) {
+        log.error(e.message)
+        //We dont have access to ProblemHandler here in boot-security. But use a problem to be consistent.
+        Problem problem = new UnexpectedProblem().cause(e).detail(e.message)
```
Comment (basejump) [inline security/boot-security/src/main/groovy/yakworks/security/spring/token/TokenController.groovy:140]: a user not being found is not an UnexpectedProblem, that is a know problem. slow down and get your arms around this so we are not doing so much back and forth here

**Review by basejump — CHANGES_REQUESTED** [review]
see comment

File: security/boot-security/src/main/groovy/yakworks/security/spring/token/TokenController.groovy (hunk 2 above)
Comment (snimavat) [reply security/boot-security/src/main/groovy/yakworks/security/spring/token/TokenController.groovy:140]: I added this as a general exception handler, will have check for specific UserNotFoundException

**Pull request merged**